packages/lockfile-file/test/read.test.ts

```diff
@@ -190,3 +191,71 @@ test('existsWantedLockfile()', async () => {
   })
   expect(await existsWantedLockfile(projectPath)).toBe(true)
 })
+
+test('readWantedLockfile() when useGitBranchLockfile', async () => {
+  getCurrentBranch.mockReturnValue('branch')
+  const lockfile = await readWantedLockfile(path.join('fixtures', '6'), {
+    ignoreIncompatible: false,
+  })
+  expect(lockfile?.importers).toEqual({
+    '.': {
+      specifiers: {
+        'is-positive': '1.0.0',
+      },
+    },
+  })
+  expect(lockfile?.packages).toStrictEqual({
+    '/is-positive/1.0.0': {
+      resolution: {
+        integrity: 'sha1-ChbBDewTLAqLCzb793Fo5VDvg/g=',
+      },
+    },
+  })
+
+  const gitBranchLockfile = await readWantedLockfile(path.join('fixtures', '6'), {
+    ignoreIncompatible: false,
+    useGitBranchLockfile: true,
+  })
+  expect(gitBranchLockfile?.importers).toEqual({
+    '.': {
+      specifiers: {
+        'is-positive': '2.0.0',
+      },
+    },
+  })
+  expect(gitBranchLockfile?.packages).toStrictEqual({
+    '/is-positive/2.0.0': {
+      resolution: {
+        integrity: 'sha1-ChbBDewTLAqLCzb793Fo5VDvg/g=',
+      },
+    },
+  })
+})
+
+test('readWantedLockfile() when useGitBranchLockfile and mergeGitBranchLockfiles', async () => {
```

Comment:
Is there a scenario in which useGitBranchLockfile is false and mergeGitBranchLockfiles is true?